src/main/java/core/mate/academy/service/ExcavatorProducer.java

```diff
@@ -0,0 +1,12 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import java.util.List;
+
+public class ExcavatorProducer implements MachineProducer<Machine> {
```

Comment:
The `ExcavatorProducer` class should implement `MachineProducer<Excavator>` instead of `MachineProducer<Machine>` to reflect the specific type of machines it produces. This ensures type safety and clarity in the code.